cpp/src/reductions/reductions.cpp

```diff
@@ -22,13 +22,23 @@ namespace cudf {
 namespace experimental {
 namespace detail {
 
+namespace {
```

Comment:
BTW, I didn't mean to suggest you should eliminate `make_default_constructed_scalar()`. I was just suggesting it could be simplified to JUST be more like what you have here -- no SFINAE needed.